version: "3"

services:
  db:
    image: postgres:latest
    env_file:
      - .env
    expose:
      - "$POSTGRES_PORT"
    volumes:
      - pg_data:/var/lib/postgresql/data
    healthcheck:
      test: ["CMD-SHELL", "-c", "pg_isready -U ${POSTGRES_USER} -d ${POSTGRES_DB}"]
      interval: 10s
      retries: 5
      timeout: 5s

  redis:
    image: redis:latest
    restart: on-failure
    expose:
      - "6379"

  app:
    build: .
    tty: true
    ports:
      - "8000:8000"
    command: bash -c "
      python manage.py migrate
      && python manage.py runserver 0.0.0.0:8000
      "
    depends_on:
      db:
        condition: service_healthy
    volumes:
      - .:/app

  celery:
    build: .
    tty: true
    command: celery -A config worker -l INFO
    restart: on-failure
    depends_on:
      - redis
      - db
      - app
    volumes:
      - .:/app
    env_file:
      - .env

  celery-beat:
    build: .
    tty: true
    command: celery -A config beat -l INFO
    restart: on-failure
    depends_on:
      - redis
      - db
      - app
    volumes:
      - .:/app
    env_file:
      - .env


volumes:
  pg_data: